src/test/java/lambda/part3/exercise/Exercise4.java

```diff
@@ -0,0 +1,124 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+class Exercise4 {
+
+    private static class LazyCollectionHelper<T, R> {
+
+        private final List<T> source;
+        private final Function<T, List<R>> toList;
+
+        LazyCollectionHelper(List <T> source, Function <T, List<R>> toList) {
+            this.source = source;
+            this.toList = toList;
+        }
+
+        public static <T> LazyCollectionHelper<T, T> from(List<T> list) {
+            return new LazyCollectionHelper <>(list, Arrays::asList);
```

Comment:
Well done. Here (and below) you can use `Collections::singletonList` instead of `Arrays::asList` and it will have the same behavior, but with a bit less memory usage.